sw/device/tests/sim_dv/chip_sw_lc_ctrl_scrap_test.c

```diff
@@ -16,95 +16,56 @@
- * `TestLocked1` state.
- * 9). Repeat the steps above in a few iterations.
+ * 2). Program LC state transition request to advance to `Scrap` state,
+ * if doing the transition through SW.
+ * 3). Return control to Host side
+ * 4). Host will wait for LC_CTRL to enter PostTransition state, then issue a
+ * hard reset 5). Host will check the LC state is `Scrap`
  *
  * In summary, this sequence walks through the following LC states:
- * "TestLocked1" -> "TestUnlocked2" -> "Dev"
+ * "Any Source State (!Scrap) -> Scrap"
  */
 
 bool execute_lc_ctrl_scrap_test(bool use_ext_clk) {
-  LOG_INFO("Start LC_CTRL scrap test");
+  if (kPerformTransitionBySW) {
+    LOG_INFO("Start LC_CTRL scrap test");
 
-  mmio_region_t lc_reg = mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR);
-  CHECK_DIF_OK(dif_lc_ctrl_init(lc_reg, &lc));
+    mmio_region_t lc_reg =
+        mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR);
+    CHECK_DIF_OK(dif_lc_ctrl_init(lc_reg, &lc));
 
-  LOG_INFO("Read and check LC state.");
-  dif_lc_ctrl_state_t curr_state;
-  CHECK_DIF_OK(dif_lc_ctrl_get_state(&lc, &curr_state));
-
-  // The OTP preload image hardcodes the initial LC state transition count to 8.
-  // With each iteration of test, there are two LC_CTRL state transitions.
-  // And the first LC_CTRL state transition is done via external JTAG interface.
-  // `kTestIterationCount` starts with 1 in SystemVerilog.
-  const uint8_t kLcStateTransitionCount = 8 + 1 + (kTestIterationCount - 1) * 2;
-
-  if (curr_state == kDifLcCtrlStateTestUnlocked2) {
-    // LC TestUnlocked2 is the intial test state for this sequence.
-    // The sequence will check if lc_count matches the preload value.
-    lc_ctrl_testutils_check_transition_count(&lc, kLcStateTransitionCount);
-
-    // Request lc_state transfer to Dev state.
-    dif_lc_ctrl_token_t token;
-    for (int i = 0; i < LC_TOKEN_SIZE; i++) {
-      token.data[i] = kLcExitToken[i];
-    }
+    // acquire the mutex and perform a transition to SCRAP
     CHECK_DIF_OK(dif_lc_ctrl_mutex_try_acquire(&lc));
     CHECK_DIF_OK(
-        dif_lc_ctrl_configure(&lc, kDifLcCtrlStateDev, use_ext_clk, &token),
+        dif_lc_ctrl_configure(&lc, kDifLcCtrlStateScrap, use_ext_clk, NULL),
         "LC transition configuration failed!");
     CHECK_DIF_OK(dif_lc_ctrl_transition(&lc), "LC transition failed!");
 
     LOG_INFO("Waiting for LC transtition done and reboot.");
-    wait_for_interrupt();
 
+    // release the access mutex to LC conroller's registers before
+    // finishing the SW side
+    CHECK_DIF_OK(dif_lc_ctrl_mutex_release(&lc));
+
+    // At this point a reset should be issued for the new LC state to take
+    // effect However, as we are transitioning into SCRAP, the core won't be
+    // able to execute any SW code, so this sequence will never return true, and
+    // the SV side will fail. In order to avoid this, we return true here and
+    // delegate the system reset task to the Host side
```

Comment:
can you add the following here instead (and update the comment?):
```c
wait_for_interrupt();
CHECK(false, "Should have reset before this line.");
```
Just to make the reset requirement a bit more explicit.

Reply:
Sure, will do!